### PR TITLE
Update to Slevomat/CS 4.7 and add some new Snifss

### PR DESCRIPTION
**Pull request by carusogabriel**
I'm not sure if we and to add [all new Snifss](https://github.com/slevomat/coding-standard/releases/tag/4.7.0).

### CODE REVIEW

File: tests/Sniffs/TestCase.php [hunk 1]
```diff
@@ -162,7 +162,7 @@ protected function assertNoSniffErrorInFile(PhpCsFile $file): void
 	private function getFormattedErrorsForFile(array $errorsForFile): string
 	{
 		$message = '';
-		foreach ($errorsForFile as $line => $errorsForPossition) {
+		foreach ($errorsForFile as $line) {
```
Comment (mhujer) [inline tests/Sniffs/TestCase.php]: I think this changes the behaviour of the code - see https://3v4l.org/XTpc9

Comment (carusogabriel) [reply tests/Sniffs/TestCase.php]: I believe we should not look for variables at foreachs than.

/cc @kukulich

**Comment by VasekPurchart** [comment]
Thank you for the PR, but I cannot accept it now, because of the [stability promise](https://github.com/consistence/coding-standard#stability) of this package - not introducing sniffs which are not covered by the written standard. I would have to release next major version which I am not ready for atm - need to add more things.

The last version of Slevomat CS has some great sniffs in it, so I had already made notes to add them to the standard in the next major, here is a temp list I made:

* SlevomatCodingStandard.Classes.ModernClassNameReference
* SlevomatCodingStandard.Functions.StaticClosure
* SlevomatCodingStandard.Operators.RequireCombinedAssignmentOperator
* SlevomatCodingStandard.TypeHints.NullTypeHintOnLastPosition
* SlevomatCodingStandard.Classes.TraitUseDeclaration
* SlevomatCodingStandard.Classes.TraitUseSpacing
* SlevomatCodingStandard.PHP.UselessSemicolon

Then there is the issue of sniffs, which are not really covering coding standard per se, but are more of static analysis nature (like `UnusedVariable` etc.). These are mentioned in the readme, but I don't think it is desirable to add them in a minor version again. In fact I was thinking about this and it might be even better to separate all these sniffs into separate ruleset (and then maybe create a combined one) - or some other option like that - I will have to investigate it before the next major release. These sniffs are of course useful, but can be redundant to tools like PHPStan etc. + PHPCS will be always more limited in this area. So this is great if you can't use it somewhere else, but I don't think they are in the same area as the other "really" coding standard sniffs.

I have prepared a new PR with new Slevomat sniffs, which are covering the existing standard: https://github.com/consistence/coding-standard/pull/50

**Comment by carusogabriel** [comment]
@VasekPurchart Thank you for this full explanation :blush:

**Pull request closed without merging**